src/rust/bitbox02/src/lib.rs

```diff
@@ -75,28 +76,24 @@ pub fn delay(duration: Duration) {
 
 // Safe wrapper for workflow_confirm
 pub fn workflow_confirm(title: &str, body: &str, longtouch: bool, accept_only: bool) -> bool {
-    // Ensure valid nullterminated C-str
-    // Will truncate title if it is too long
-    let title_cstr = {
-        const TITLE_LEN: usize = 20;
-        let len = core::cmp::min(TITLE_LEN, title.len());
-        let mut buf = [0u8; TITLE_LEN + 1];
-        // resize title to actual length
-        let title = &title.as_bytes()[0..len];
-        // copy from title to buf
-        buf[0..len].copy_from_slice(title);
-        buf
+    // Create null-terminated strings for title and body
+    let title_cstr = match str_to_cstr!(title, 20) {
+        Ok(cstr) => cstr,
+        Err(()) => {
+            unsafe {
+                bitbox02_sys::screen_print_debug("string didn't fit\0".as_ptr() as *const _, 3000)
+            };
+            return false;
+        }
     };
-    // same as title_cstr
-    let body_cstr = {
-        const BODY_LEN: usize = 100;
-        let len = core::cmp::min(BODY_LEN, body.len());
-        let mut buf = [0u8; BODY_LEN + 1];
-        // resize body to actual length
-        let body = &body.as_bytes()[0..len];
-        // copy from body to buf
-        buf[0..len].copy_from_slice(body);
-        buf
+    let body_cstr = match str_to_cstr!(body, 100) {
+        Ok(cstr) => cstr,
+        Err(()) => {
+            unsafe {
+                bitbox02_sys::screen_print_debug("string didn't fit\0".as_ptr() as *const _, 3000)
```

Comment:
I don't really want to have it as part of the macro. Since `screen_print_debug` actually prints to the screen I prefer to have it in the "logic" part of the code.

Reply:
I'll create a wrapper for `screen_print_debug` and make it look a little prettier.